compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -2406,7 +2406,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
     }
 
     fn lower_stmt(&mut self, s: &Stmt) -> SmallVec<[hir::Stmt<'hir>; 1]> {
-        let kind = match s.kind {
+        let (hir_id, kind) = match s.kind {
```

Comment:
I'm not sure I understand the purpose of this commit (aefbe667). Could you describe a bit what's going on?

Reply:
The reason is this line: https://github.com/rust-lang/rust/blob/master/compiler/rustc_hir/src/hir.rs#L1162
HIR does not make a difference between attributes on a statement and on the statement's content.
I reproduce this aliasing in the `HirId -> &[Attributes]` array.